javaagent-extension-api/src/main/java/io/opentelemetry/javaagent/extension/StructuredConfigPropertiesBridge.java

```diff
@@ -0,0 +1,153 @@
+import io.opentelemetry.sdk.autoconfigure.spi.internal.StructuredConfigProperties;
+import java.time.Duration;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.BiFunction;
+import javax.annotation.Nullable;
+
+/**
+ * A {@link ConfigProperties} which resolves properties based on {@link StructuredConfigProperties}.
+ *
+ * <p>Only properties starting with "otel.instrumentation." are resolved. Others return null (or
+ * default value if provided).
+ *
+ * <p>To resolve:
+ *
+ * <ul>
+ *   <li>"otel.instrumentation" refers to the ".instrumentation.java" node
+ *   <li>The portion of the property after "otel.instrumentation." is split into segments based on
+ *       ".".
+ *   <li>For each N-1 segment, we walk down the tree to find the relevant leaf {@link
+ *       StructuredConfigProperties}.
+ *   <li>We extract the property from the resolved {@link StructuredConfigProperties} using the last
+ *       segment as the property key.
+ * </ul>
+ *
+ * <p>For example, given the following YAML, asking for {@code
+ * ConfigProperties#getString("otel.instrumentation.common.string_key")} yields "value":
+ *
+ * <pre>
+ *   instrumentation:
+ *     java:
+ *       common:
+ *         string_key: value
+ * </pre>
+ */
+final class StructuredConfigPropertiesBridge implements ConfigProperties {
+
+  private static final String OTEL_INSTRUMENTATION_PREFIX = "otel.instrumentation.";
+
+  // The node at .instrumentation.java
+  private final StructuredConfigProperties instrumentationJavaNode;
+
+  StructuredConfigPropertiesBridge(StructuredConfigProperties rootStructuredConfigProperties) {
+    instrumentationJavaNode =
+        rootStructuredConfigProperties
+            .getStructured("instrumentation", empty())
+            .getStructured("java", empty());
+  }
+
+  @Nullable
+  @Override
+  public String getString(String propertyName) {
+    return getPropertyValue(propertyName, StructuredConfigProperties::getString);
+  }
+
+  @Nullable
+  @Override
+  public Boolean getBoolean(String propertyName) {
+    return getPropertyValue(propertyName, StructuredConfigProperties::getBoolean);
+  }
+
+  @Nullable
+  @Override
+  public Integer getInt(String propertyName) {
+    return getPropertyValue(propertyName, StructuredConfigProperties::getInt);
+  }
+
+  @Nullable
+  @Override
+  public Long getLong(String propertyName) {
+    return getPropertyValue(propertyName, StructuredConfigProperties::getLong);
+  }
+
+  @Nullable
+  @Override
+  public Double getDouble(String propertyName) {
+    return getPropertyValue(propertyName, StructuredConfigProperties::getDouble);
+  }
+
+  @Nullable
+  @Override
+  public Duration getDuration(String propertyName) {
+    Long millis = getPropertyValue(propertyName, StructuredConfigProperties::getLong);
+    if (millis == null) {
+      return null;
+    }
+    return Duration.ofMillis(millis);
+  }
```

Comment:
is this intentionally scoping down and not handling durations with units like `DefaultConfigProperties`?

Reply:
Yes. All durations in the declarative config schema are integer types assumed to be ms. There is no ability to specify the `1s`, `1h`, `1ms` at this time.